### PR TITLE
Cli: enable json output

### LINKED ISSUE #7827: Make cli commands support emitting JSON
#### Problem

We have been working on a [Solana validator dashboard ](https://stakingfacilities.com/solana-validators). To offer more insights we would like to integrate information provided by cli commands like `solana validator-info get` and `solana show-block-production`. It would be extremely helpful if cli commands would support emitting JSON.

#### Proposed Solution

Add a `--json` flag to cli commands that allow programmatic access by outputting JSON.

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Some of the cli commands do a lot of client-side processing that make them valuable for users, but the displays are geared toward human eyes only. It is very difficult to ingest this data programmatically.

#### Summary of Changes
- Add global `--output` format arg, to set OutputFormat -- currently supports `Display` (default) and `Json`
- Implement json output for various process-heavy cli subcommands. All of these subcommands feature ad-hoc printing, so in order to make json serialization easy and data-maintenance sane, collect the data to return in a Cli-specific struct. Then impl Display for the struct to collect/contain human-readable print formatting.
As a result, once the struct is populated, output format-selection and printing is triggered with a single line:
`OutputFormat.formatted_print(&<STRUCT>);`

Fixes #7827

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
@mvines , I would love your feedback on this approach. It's quite a bit of churn, but seems reasonably maintainable, given all the special-case printing going on in the cli.

**Comment by mvines** [comment]
This approach looks great.   I like that we'll be able to more easily unit test the business logic of each command now as well.  

Probably want to get this landed ASAP to avoid rebase hell.   I'd backport to v1.1 to keep us sane through the end of May, but skip the v1.0 backport since that'll only live for another couple weeks

File: cli/src/cli_output.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    where
+        T: Serialize + fmt::Display,
+    {
+        match self {
+            OutputFormat::Display => {
+                println!("{}", item);
+            }
+            OutputFormat::Json => {
+                println!("{}", serde_json::to_value(item).unwrap());
+            }
```
[the remark this answers is not in the document]
Comment (mvines) [reply cli/src/cli_output.rs:38]: default to `to_string_pretty()`?

[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply cli/src/cli_output.rs:42]: Cool, Json default is now `to_string_pretty()`. I added a JsonCompact variant to provide the former behavior.

[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply cli/src/cli_output.rs:42]: Thanks, @mvines and @t-nelson !

**Pull request merged**